data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.gateway/1.0/corda.p2p.gateway.json

```diff
@@ -0,0 +1,102 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://corda.r3.com/net/corda/schema/configuration/p2p/1.0/corda.p2p.gateway.json",
+  "title": "Corda P2P Gateway Configuration Schema",
+  "description": "Configuration schema for the P2P gateway.",
+  "type": "object",
+  "properties": {
+    "hostAddress": {
+      "description": "The host name or IP address where the HTTP server will bind.",
+      "type": "string",
+      "default": "0.0.0.0"
+    },
+    "hostPort": {
+      "description": "The HTTP port.",
+      "type": "integer",
+      "default": 0,
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "sslConfig": {
+      "description": "Used for one-way TLS between Gateways.",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "revocationCheck": {
+          "type": "object",
+          "default": {},
+          "description": "How to check for revocation of the server certificate.",
+          "properties": {
+            "mode": {
+            "oneOf": [
+              {"const": "SOFT_FAIL", "title": "The revocation check will succeed if the revocation status cannot be determined under certain circumstances. See java.security.cert.PKIXRevocationChecker.Option.SOFT_FAIL java doc for more details."},
+              {"const": "HARD_FAIL", "title": "Opposite of SOFT_FAIL - i.e. most rigorous check. Among other things, this check requires that CRL checking URL is available on every level of certificate chain."},
+              {"const": "OFF", "title": "Switch CRL check off."}
+            ],
+            "default": "OFF"
```

Comment:
I think we should switch the default to `SOFT_FAIL` (I also had a discussion with Matthew and he told me that's the case for C4). I think we had put `OFF` as the default in the beginning to ensure the default setup without CRLs would work, but this should be the case with `SOFT_FAIL`, while also using them when there.

Reply:
I will do that and run a test with the p2p deployment.